From 91b993c36b599cbdd8d3ba7a781e018eb3e20477 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 14 Jun 2022 05:10:57 +0300
Subject: [PATCH] [SW-89971] rom: add twi if handler in recovery

As a part of recovery config the twi interface handler is added to
configure the interface enable/disable and axprot values. This is
done additionaly in recovery flow in order to make sure of the
interface state pre range register configuration.

Change-Id: Ic45222cef59314044b9a0f2bbfd4c4426c361847
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 9ac7779..323e02b 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -743,6 +743,8 @@
 
 void security_rr_recovery_config(void)
 {
+	security_handle_twi();
+
 	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
 	security_rr_config_lbw_if_common(&rr_ctx_lbw[RR_OPEN_ALL]);
 
